cpp/CMakeLists.txt

```diff
@@ -244,51 +285,150 @@ add_library(cugraph SHARED
-    PURPOSE "Google C++ Testing Framework (Google Test)."
-    URL "https://github.com/google/googletest")
-
+# - link libraries --------------------------------------------------------------------------------
 
+target_link_libraries(cugraph PRIVATE
+    ${CUDF_LIBRARY} ${RMM_LIBRARY} cublas cusparse curand cusolver cudart cuda ${Boost_LIBRARIES})
+if(OpenMP_CXX_FOUND)
+target_link_libraries(cugraph PRIVATE
 ###################################################################################################
-# - Generate the Tests -------------------------------------------------------------------------------
-if(GTEST_FOUND)
-    message(STATUS "Google C++ Testing Framework (Google Test) found in ${GTEST_ROOT}")
-    include_directories(${GTEST_INCLUDE_DIRS})
-    add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/src/tests)
-else()
-    message(AUTHOR_WARNING "Google C++ Testing Framework (Google Test) not found: automated tests are disabled.")
-endif()
+###   Use ${OpenMP_CXX_LIB_NAMES} instead of OpenMP::OpenMP_CXX to avoid the following warnings.
+###
+###   Cannot generate a safe runtime search path for target TARGET_NAME
+###   because files in some directories may conflict with libraries in implicit
+###   directories:
+###   ...
+###
+###   libgomp.so is included in the conda base environment and copied to every new conda
+###   environment. If a full file path is provided (e.g ${CUDF_LIBRARY} and ${RMM_LIBRARY}), cmake
+###   extracts the directory path and adds the directory path to BUILD_RPATH (if BUILD_RPATH is not
+###   disabled).
+###
+###   cmake maintains a system specific implicit directories (e.g. /lib, /lib/x86_64-linux-gnu,
+###   /lib32, /lib32/x86_64-linux-gnu, /lib64, /lib64/x86_64-linux-gnu, /usr/lib,
+###   /usr/lib/gcc/x86_64-linux-gnu/7, /usr/lib/x86_64-linux-gnu, /usr/lib32,
+###   /usr/lib32/x86_64-linux-gnu, /usr/lib64, /usr/lib64/x86_64-linux-gnu,
+###   /usr/local/cuda-10.0/lib64", /usr/local/cuda-10.0/lib64/stubs).
+###
+###   If a full path to libgomp.so is provided (which is the case with OpenMP::OpenMP_CXX), cmake
+###   checks whether there is any other libgomp.so with the different full path (after resolving
+###   soft links) in the search paths (implicit directoires + BUILD_RAPTH). There is one in the
+###   path included in BUILD_RPATH when ${CUDF_LIBRARY} and ${RMM_LIBRARY} are added; this one can
+###   potentially hide the one in the provided full path and cmake generates a warning (and RPATH
+###   is searched before the directories in /etc/ld.so/conf; ld.so.conf does not coincide but
+###   overlaps with implicit directories).
+###
+###   If we provide just the library names (gomp;pthread), cmake does not generate warnings (we
+###   did not specify which libgomp.so should be loaded in runtime), and the one first found in
+###   the search order is loaded (we can change the load library by setting LD_LIBRARY_PATH or
+###     manually editing BUILD_RPATH).
```

Comment:
Are there cases where a developer/user would want to manually set it? If so, should we document this in CONTRIBUTING.md and expose a cmake variable?

Reply:
See the answer for 1 for your previous question. I guess it will rarely be a case (unless we are using new OpenMP features and conda comes with old OpenMP library), but I included the instruction in comments for the case this actually becomes a problem (it took me a quite good amount of effort to figure this out, and I wanted to properly document this in a place we can easily find once this becomes necessary).

Reply:
Ok, that sounds safe enough. Let's keep it like this for now. The documentation is very useful, we can quickly provide a more advanced solution if needed.